dkg/pedersen/mod.go

```diff
@@ -196,10 +201,65 @@ func (a *Actor) Encrypt(message []byte) (K, C kyber.Point, remainder []byte,
 	return K, C, remainder, nil
 }
 
+// VerifiableEncrypt implements dkg.Actor. It uses the DKG public key to encrypt
+// a message and provide a zero knowledge proof that the encryption is done by
+// this person.
+//
+// See https://arxiv.org/pdf/2205.08529.pdf / section 5.4 Protocol / step 1
+func (a *Actor) VerifiableEncrypt(message []byte, GBar kyber.Point) (ciphertext types.Ciphertext, remainder []byte,
```

Comment:
This looks very similar to `checkEncryptionProof` - one more reason to have a `kyber` library.